### DISCUSSION TITLE
Running inferences backward in eval.py?

### DISCUSSION THREAD

**Discussion by jama1017** [hkchengrex/XMem#65] [Q&A] [answered]
Hi!

In the interactive demo, we can choose to propagate a mask backwards, which can be very helpful for some shots where an object is first zoomed in and then slowly zoomed out. Can we do this in eval.py? (I guess a manual way to do this is to process the video into frames backward, but I just wonder if there is an integrated way)

**Comment by hkchengrex** [comment, marked as answer]
It is not implemented.

**Reply by hkchengrex** [reply]
@jama1017 Might be easy to do by just reversing the sorted lists in 
https://github.com/hkchengrex/XMem/blob/af90122693a7f053b62bbeee3c2cc90d43b541ac/inference/data/video_reader.py#L38-L40

**Reply by jama1017** [reply]
Ah that's a good tip. Thanks so much!
